### PR TITLE
Rework JetStream delivery policy

### LINKED ISSUE #2314: Rework JetStream deliver policy
Current NATS JetStream metadata is not in line with most NATS clients regarding delivery policies, and on top of that, does not provide all the options.

Right now it has:
`deliverAll: bool`
which drives deliveryPolicy based on true/false and startTime/startSequence.
This does not allow to use `deliver policy new`.
## Describe the feature
Reword JetStream Metada to be in line with nats.go client:

```diff
- deliverAll: bool
+ deliverPolicy: nats.DeliverPolicy
```
**This will be a breaking change**

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by Jarema**
Signed-off-by: Tomasz Pietrek <tomasz@nats.io>

# Description

_Please explain the changes you've made_

## Issue reference

closes #2314
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pubsub/jetstream/metadata.go [hunk 1]
```diff
@@ -144,6 +141,22 @@ func parseMetadata(psm pubsub.Metadata) (metadata, error) {
...
+	case "last":
+		m.deliverPolicy = nats.DeliverLastPolicy
+	case "new":
+		m.deliverPolicy = nats.DeliverNewPolicy
+	case "sequence":
+		m.deliverPolicy = nats.DeliverByStartSequencePolicy
+	case "time":
+		m.deliverPolicy = nats.DeliverByStartTimePolicy
+	default:
+		return metadata{}, fmt.Errorf("deliver policy %s is not one of: all, last, next", deliverPolicy)
```
Comment (piotrpio) [inline pubsub/jetstream/metadata.go:157]:
```suggestion
		return metadata{}, fmt.Errorf("deliver policy %s is not one of: all, last, new, sequence, time", deliverPolicy)
```

**Review by piotrpio — CHANGES_REQUESTED** [review]

File: pubsub/jetstream/metadata.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"time"
 
 	"github.com/dapr/components-contrib/pubsub"
+	"github.com/nats-io/nats.go"
```
Comment (berndverst) [inline pubsub/jetstream/metadata.go:23]: Your imports need to be rearranged.

there need to be 3 blocks of imports in the following order:

- standard library
- other packages
- dapr packages (github.com/dapr)

**Review by berndverst — CHANGES_REQUESTED** [review]
just fix the import order and then this looks good to me

**Review by ItalyPaleAle — DISMISSED** [review]
LGTM

Please also open a PR in the docs repo (https://github.com/dapr/docs/) to document the added/removed metadata options (in the 1.10 branch)

**Comment by Jarema** [comment]
docs PR added: https://github.com/dapr/docs/pull/3007

**Review by piotrpio — APPROVED** [review]
LGTM

**Review by yaron2 — APPROVED** [review]

**Pull request merged**